### PR TITLE
Add travis

### PR DESCRIPTION
**Pull request by stbenjam**
This brings 896 failures to 0.  There are dozens of broken links across the web site.

### CODE REVIEW

**Comment by stbenjam** [comment]
The first two commits are https://github.com/Katello/katello.org/pull/117, https://github.com/Katello/katello.org/pull/119 which are needed to get tests to pass

**Comment by ehelms** [comment]
Do you think we should test the deploy script?

**Comment by stbenjam** [comment]
Um, it's tricky.  I had only planned to backport this to 2.1, 2.0 tests are going to fail even worse.

It will also need some weird juggling in the travis job, to find out the target branch from the GitHub API, merge the PR to it, checkout deploy, run the job, etc.  Something like...


```ruby
if (pr = ENV['TRAVIS_PULL_REQUEST']) && !pr.empty?
    puts "Testing PR #{pr}"
    JSON.load(open("https://api.github.com/repos/katello/Katello.org/pulls/#{pr}").read)
    sh "git checkout #{pr_details['base']['ref']}"
    sh "git fetch origin +refs/pull/#{pr}/merge"
    sh 'git merge FETCH_HEAD'
    sh 'git checkout deploy'
     # etc, etc
end
```

It's *totally* possible, but because this fixes so many broken things now, I'd rather we take it against nightly, and make it another rainy day thing?

**Comment by stbenjam** [comment]
Well, I guess I could have it ignore 2.0.  Maybe it's possible, I'll try it

**Comment by stbenjam** [comment]
Ah, not so bad actually. It will need a little bit of jugging, this has to get merged, then backported to 2.1...and then tests will pass.

**Comment by stbenjam** [comment]
Proof tests pass (running against my branch): https://travis-ci.org/Katello/katello.org/builds/49631765

**Comment by stbenjam** [comment]
[test]

**Comment by stbenjam** [comment]
@ehelms I need https://github.com/Katello/katello.org/pull/121 for this, otherwise I can never test a PR because deploy.rb always will build from the github repo.

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'json'
+require 'open-uri'
+require 'html/proofer'
+
+namespace :test do
+  task :travis do
+    if (pr = ENV['TRAVIS_PULL_REQUEST']) && !pr.empty?
+      sh 'git clone https://github.com/stbenjam/katello.org.git -b customize-path deploy'
```
Comment (stbenjam) [inline Rakefile]: Will change to this when https://github.com/Katello/katello.org/pull/121 is merged:

```
sh 'git clone https://github.com/Katello/katello.org.git -b deploy deploy'
```

File: Rakefile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+      Dir.chdir 'deploy'
+      sh "./deploy.rb --pr #{pr}"
+      Dir.chdir '..'
```
Comment (ehelms) [inline Rakefile]: You can do a block here and avoid the change back, e.g.

```
Dir.chdir('deploy') do
  blah blah
end
```

Comment (stbenjam) [inline Rakefile]: @ehelms Neat, updated

**Comment by ehelms** [comment]
ACK

**Pull request merged**